backend-project/small_eod/letters/views.py

```diff
@@ -37,6 +46,37 @@ class LetterViewSet(viewsets.ModelViewSet):
         "modified_by__username",
     ]
 
+    def get_queryset(self):
+        if self.action == "list":
+            return (
+                Letter.objects.annotate(attachments_count=Count("attachments"))
```

Comment:
Z mojego doświadczenia w widokach złożone zapytania długoterminowo są kosztowne w utrzymaniu i skłaniają do nadmiernych powtórzeń. Idealnie, gdy nie masz żadnych importów do `models.Count`, `models.Prefetch`, `models.Q` i innych "niskopoziomowych" konstruktorów zapytań w widoku, a enkapsulujesz to w modelu. 

Ja proponuje przenieść to do `LetterQuerySet`, a w tym miejscu zapisać `Letter.objects.with_attachment_count().with_list_related().all()`. Zwracam uwagę, że tego queryset będziesz potrzebować także, aby w dowolnym innym kontekście efektywnie wykorzystać `LetterListSerializer`, który z założenia powininien być otwarty na ponowne wykorzystanie.